compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -630,13 +630,9 @@ public void visit(BLangClassDefinition classDefinition, AnalyzerData data) {
     }
 
     private void analyzeClassDefinition(BLangClassDefinition classDefinition, AnalyzerData data) {
-        SymbolEnv currentEnv = data.env;
-        SymbolEnv classEnv = SymbolEnv.createClassEnv(classDefinition, classDefinition.symbol.scope, currentEnv);
         for (BLangSimpleVariable field : classDefinition.fields) {
-            data.env = classEnv;
             analyzeDef(field, data);
```

Comment:
By doing this, when analyzing fields of objects we set the scope as package scope. 
But in symbol enter when we define these fields we create new scope(for classes) and define these fields within that scope. So IMO here also we need to create that scope.

Reply:
Simply, what I want to do is that when analyzing a field, that field should not be able to see previous fields. I don't think we need to create a new scope even if we did in SymbolEnter.

Reply:
We need. if not we are going to define fields of class within the package scope.

Reply:
Can we find any failing scenario?